etest_test/profile_test.py

```diff
@@ -0,0 +1,221 @@
+    docker_arch = "ppc64le"
+
+    result = sut.Profile(logger, False, arch, "glibc", False, True, False)
+
+    # Check profile strings
+    assert result.profile == arch
+    assert result.docker == docker_arch
+
+    # Check arch values
+    assert result.arch == arch
+    assert result.pkg_arch == "ppc64"
+    assert result.docker_arch == docker_arch
+
+    # Check profile options
+    assert result.hardened is False
+    assert result.multilib is True
+    assert result.systemd is False
+
+    # Check libc
+    assert result.libc == "glibc"
+
+
+def test_base_arm() -> None:
+    """Ensure ARM base arches work."""
+    docker_arm_mappings = {"armv5": "armv5tel", "armv6": "armv6j_hardfp", "armv7": "armv7a_hardfp"}
+
+    for arch in ["armv5", "armv6", "armv7"]:
+        result = sut.Profile(logger, False, arch, "glibc", False, True, False)
+
+        # Check profile strings
+        assert result.profile == arch
+        assert result.docker == docker_arm_mappings[arch]
+
+        # Check arch values
+        assert result.arch == arch
+        assert result.pkg_arch == "arm"
+        assert result.docker_arch == docker_arm_mappings[arch]
+
+        # Check profile options
+        assert result.hardened is False
+        assert result.multilib is True
+        assert result.systemd is False
+
+        # Check libc
+        assert result.libc == "glibc"
+
+
+def test_libc() -> None:
+    """Ensure alternative libcs work."""
+    arch = "amd64"
+
+    for libc in ["musl", "uclibc"]:
+        result_vanilla = sut.Profile(logger, False, arch, libc, False, True, False)
+        result_hardened = sut.Profile(logger, False, arch, libc, True, True, False)
+
+        # Check profile strings
+        assert result_vanilla.profile == f"{arch}-{libc}-vanilla"
+        assert result_vanilla.docker == f"{arch}-{libc}-vanilla"
+        assert result_hardened.profile == f"{arch}-{libc}-hardened"
+        assert result_hardened.docker == f"{arch}-{libc}-hardened"
+
+        # Check arch values
+        assert result_vanilla.arch == result_hardened.arch == arch
+        assert result_vanilla.pkg_arch == result_hardened.pkg_arch == arch
+        assert result_vanilla.docker_arch == result_hardened.docker_arch == arch
+
+        # Check profile options
+        assert result_vanilla.hardened is False
+        assert result_hardened.hardened is True
+        assert result_vanilla.multilib is True
+        assert result_hardened.multilib is True
+        assert result_vanilla.systemd is False
+        assert result_hardened.systemd is False
+
+        # Check libc
+        assert result_vanilla.libc == result_hardened.libc == libc
+
+
+def try_invalid_profile(*args: Any, **kwargs: Any) -> bool:
+    """Try an invalid profile."""
+    e = False
+
+    try:
+        sut.Profile(*args, **kwargs)
+    except sut.InvalidProfileError:
+        e = True
+
+    return e
+
+
+def test_invalid_arm_profiles() -> None:
+    """Ensure invalid ARM profiles raise exceptions."""
```

Comment:
These are just parametrised cases again.

In fact, because this space is so much larger, consider using hypothesis to generated these to get a much better sampling of the invalid space.